src/main/java/com/hartwig/snpcheck/SnpCheck.java

```diff
@@ -74,52 +90,83 @@ public void handle(final PipelineComplete event) {
-                    .map(Page::iterateAll)
-                    .orElse(Collections.emptyList());
-            Optional<Sample> maybeRefSample = onlyOne(samples, run.getSet(), SampleType.REF);
-            Optional<Sample> maybeTumorSample = onlyOne(samples, run.getSet(), SampleType.TUMOR);
-            if (maybeRefSample.isPresent()) {
-                Sample refSample = maybeRefSample.get();
-                Optional<Blob> maybeValVcf = findValidationVcf(valVcfs, refSample);
-                if (maybeValVcf.isPresent()) {
-                    Result result = doComparison(run, refSample, maybeValVcf.get());
-                    SnpCheckEvent.builder()
-                            .publisher(turquoiseTopicPublisher)
-                            .sample(maybeTumorSample.map(Sample::getName).orElse(refSample.getName()))
-                            .result(result.name().toLowerCase())
-                            .build()
-                            .publish();
-                    if (result.equals(Result.PASS)) {
-                        publishValidated(event);
-                    }
-                } else {
-                    LOGGER.info("No validation VCF available for set [{}].", run.getSet().getName());
-                    labPendingBuffer.add(event);
-                }
-            } else {
-                LOGGER.warn("Set [{}] had no ref sample available in the API. Unable to locate validation VCF.",
-                        run.getSet().getName());
-                apiFailed(run, TypeEnum.TECHNICALFAILURE);
-            }
-        } else {
+        if (run.getStatus() != Status.FINISHED && !runFailedQc(run)) {
             LOGGER.info("Skipping run with status [{}]", run.getStatus());
+            return;
+        }
+        Iterable<Blob> valVcfs = Optional.ofNullable(pipelineStorage.list(bucketName, Storage.BlobListOption.prefix(SNPCHECK_VCFS)))
+                .map(Page::iterateAll)
+                .orElse(Collections.emptyList());
+        var runSet = run.getSet();
+        Optional<Sample> maybeRefSample = findSample(runSet, SampleType.REF);
+        Optional<Sample> maybeTumorSample = findSample(runSet, SampleType.TUMOR);
+        if (maybeRefSample.isEmpty()) {
+            LOGGER.warn("Set [{}] had no ref sample available in the API. Unable to locate validation VCF.", runSet.getName());
+            apiFailed(run, TypeEnum.TECHNICALFAILURE);
+            return;
         }
+        Sample refSample = maybeRefSample.get();
+        Optional<Blob> maybeValVcf = findValidationVcf(valVcfs, refSample);
+        if (maybeValVcf.isEmpty()) {
+            LOGGER.info("No validation VCF available for runSet [{}].", runSet.getName());
+            labPendingBuffer.add(event);
+            return;
+        }
+        Result result = doComparison(run, refSample, maybeValVcf.get());
+        publishTurquoiseEvent(maybeTumorSample.map(Sample::getName).orElse(refSample.getName()), result.name());
```

Comment:
Why does snpcheck also publish Turquoise events? Isn't the idea to move entirely to Aqua?